cilium/CFP-25694-move-cilium-cli-to-cilium-repo.md

```diff
@@ -0,0 +1,145 @@
+repository is to minimize the feature development overhead for adding new tests
+to `cilium connectivity test`.
+
+Currently, adding a new test to `cilium connectivity test` requires the following
+steps:
+
+- Open a pull request against [cilium/cilium-cli] repo to add a new test, and
+  wait for the CI build to finish.
+- Modify GitHub workflows files to pick up the CI version of `cilium-cli` in
+  your pull request against [cilium/cilium] repo.
+- Once the new tests passes, undo the change in the workflow files.
+- Get these pull requests reveiewed and merged.
+
+Even after all these steps, the new test does not become a part of the regular
+CI until the [sig-cilium-cli] team releases a new version of `cilium-cli` that
+includes the test, and Renovate updates GitHub wokflow files to pick up the new
+`cilium-cli` version. This has been a major source of frustration for some Cilium
+contributors.
+
+## Goals
+
+* Move [cilium/cilium-cli] code into [cilium/cilium] main branch under `cli/`
+  directory. This enables Cilium contributors to implement new features with
+  tests in a single pull request against [cilium/cilium] repository.
+* Establish a backporting process for changes inside `cli/` directory.
+* Avoid disrupting the cilium-cli development and release processes as much as
+  possible.
+
+## Non-Goals
+
+* Rename the name of the `cilium-cli` binary from `cilium` to something else.
+* Release `cilium-cli` as a part of Cilium releases.
+
+## Proposal
+
+This proposal consists of three steps to move [cilium/cilium-cli] code into
+[cilium/cilium]:
+
+### 1. Move [cilium/cilium-cli] code to [cilium/cilium] under `cli/` directory
+
+I'm assuming that you want to preserve the Git history of [cilium/cilium-cli]. See
+[michi-covalent/cilium] on how the code migration might look like. Squashing
+these commits into a single commit is another option.
+
+### 2. Make [cilium/cilium-cli] code a part of `github.com/cilium/cilium` Go module
+
+Merge [cilium/cilium-cli] code into `github.com/cilium/cilium` Go module by
+removing `cli/go.mod`. This allows [cilium/cilium-cli] code to depend on
+[cilium/cilium] code in the same tree.
+
+This is **not** to suggest `cilium-cli` to be tied to a specific `cilium`
+version. The goal of this proposal is to make it easy for contributors to add
+tests against the latest Cilium code. Continue to maintain the backward
+compatibility of `cilium-cli` with respect to `cilium` versions.
+
+### 3. Release `cilium-cli` from [cilium/cilium-cli] repository
+
+We do not need to change the release process / cadence to accomplish the goals
+defined in this CFP. Continue releasing `cilium-cli` from [cilium/cilium-cli] at
+its own cadence using its own versioning that is independent of [cilium/cilium]
+versioning as you have been doing in [cilium/cilium-cli] repository.
+
+See [michi-covalent/cilium-cli] as an example of how [cilium/cilium-cli] repo
+might look like after the code migration. Basically it contains:
```

Comment:
Another option could just be a git mirror for the cli/ directory, something like how https://github.com/libbpf/libbpf mirrors in from the Linux kernel tree.

Reply:
yeah @tklauser also suggested something similar.

it's a bit more complicated for Go modules because we need to modify all the import paths from `github.com/cilium/cilium/cli` back to `github.com/cilium/cilium-cli`.